python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -171,6 +178,25 @@ def dimensions(self) -> DimensionUniverse:
                 cache.dimensions = universe
             return cache.dimensions
 
+    @property
+    def datastore_cache_manager(self) -> AbstractDatastoreCacheManager:
+        """Cache manager to use when reading files from the butler."""
+        # RemoteButler does not get any cache configuration from the server.
+        # Read the Datastore default config (which is a FileDatastore)
+        # and obtain the default cache manager configuration. Overrides will
+        # come from environment variables. This will not work properly if the
+        # defaults for DatastoreConfig no longer include the cache.
+        if self._datastore_cache_manager is None:
+            datastore_config = DatastoreConfig()
+            self._datastore_cache_manager: AbstractDatastoreCacheManager
+            if "cached" in datastore_config:
+                self._datastore_cache_manager = DatastoreCacheManager(
```

Comment:
I've been assuming that services will set the relevant environment variables like we are going to be doing for user's notebooks. Turning off caching through environment variables can be done. We have the option of defining a separate cache manager config model that we configure to be disabled (ie keep 0 datasets) which would allow environment variables to automatically turn it on if needed. If we always use the disabled cache manager for the factory we make it impossible for anyone to use the environment to tune.

Reply:
I didn't see an environment variable that could be used to disable the cache... which one is it?

It doesn't need to be force-disabled when you instantiate via the factory -- just default disabled if there is no explicit configuration provided.

I don't think it's good to require all consumers to remember to set an environment variable in order for their stuff not to crash out... it would be better if it did the safe thing by default.  If we do want to require everyone to set an envvar then it would be good for this to be accompanied by a Phalanx PR setting it everywhere it is required.

Reply:
Currently asking for 0 datasets will effectively turn it off but I do need to add a "disabled" mode that can be used to make it explicitly disabled regardless of other settings. 

Sounds like we want to make the RemoteButler take the cache manager as a parameter and in the factory I can create it with a default disabled config but using DatastoreCacheManager so that environment variables can still be configured and respected. If it's not passed in by the constructor the code we have here can trigger so that client users can benefit from the cache.

Reply:
Yeah that sounds OK to me, though I do wonder if we should wait to add the option to enable it for factory-instantiated RemoteButlers until someone asks for it.  I can't think of a situation where this cache in its current form would be useful for a service -- I suspect we might end up writing a different implementation anyway, if someone needs it.